xcp_d/utils/hcp2fmriprep.py

```diff
@@ -177,157 +322,106 @@ def hcpfmriprepx(hcp_dir, out_dir, sub_id):
-            + "_acq-"
-            + idx[2]
-            + "_space-MNI152NLin6Asym_desc-preproc_bold.json"
-        )
-        ddjsonfile = out_dir + "/dataset_description.json"
-        ciftijson = (
-            funcdir
-            + "/sub-"
-            + sub_id
-            + "_task-"
-            + idx[1]
-            + "_acq-"
-            + idx[2]
-            + "_space-fsLR_den-91k_bold.dtseries.json"
+        # write out the confounds
+        regressors_file_base = f"{sub_ent}_{task_ent}_{dir_ent}_desc-confounds_timeseries"
+        regressors_tsv_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{regressors_file_base}.tsv",
         )
+        regressors.to_csv(regressors_tsv_fmriprep, index=False, sep="\t", na_rep="n/a")
 
-        ddjson = {"Name": "HCP", "DatasetType": "derivative", "GeneratedBy": [{"Name": "HCP"}]}
-
-        writejson(jsontis, boldjson)
-        writejson(json2, ciftijson)
-        writejson(ddjson, ddjsonfile)
-
-        # get the files
-        rawfiles = [hcp_ref, ciftip]
-        newfiles = [prep_ref, ciftib]
-
-        for kk, jj in zip(rawfiles, newfiles):
-            copyfileobj_example(kk, jj)
-            print("finished converting" + kk)
+        # NOTE: Is this JSON any good?
+        regressors_json_fmriprep = os.path.join(
+            func_dir_fmriprep,
+            f"{regressors_file_base}.json",
+        )
+        regressors.to_json(regressors_json_fmriprep)
+
+    print("finished collecting func files")
+
+    # Copy HCP files to fMRIPrep folder
+    for file_orig, files_fmriprep in copy_dictionary.items():
+        if not isinstance(files_fmriprep, list):
+            raise ValueError(
+                f"Entry for {file_orig} should be a list, but is a {type(files_fmriprep)}"
+            )
+
+        if len(files_fmriprep) > 1:
+            print(f"File used for more than one output: {file_orig}")
+
+        for file_fmriprep in files_fmriprep:
+            copyfileobj_example(file_orig, file_fmriprep)
+    print("finished copying files")
+
+    # Write the dataset description out last
+    dataset_description_dict = {
+        "Name": "HCP",
+        "DatasetType": "derivative",
+        "GeneratedBy": [
+            {
+                "Name": "HCP",
+                "Version": "unknown",
+                "CodeURL": "https://github.com/Washington-University/HCPpipelines",
+            },
+        ],
+    }
+    dataset_description_fmriprep = os.path.join(out_dir, "dataset_description.json")
+    if not os.path.isfile(dataset_description_fmriprep):
+        writejson(dataset_description_dict, dataset_description_fmriprep)
+
+    # Write out the mapping from HCP to fMRIPrep
+    scans_dict = {}
+    for key, values in copy_dictionary.items():
+        for item in values:
+            scans_dict[item] = key
+
+    # Make sure the fake files are represented accurately
+    scans_dict[template_to_t1w_orig] = "NA"
+    scans_dict[t1w_to_template_orig] = "NA"
```

Comment:
```suggestion
    scans_dict[template_to_t1w_orig] = "n/a"
    scans_dict[t1w_to_template_orig] = "n/a"
```
I like this, but "n/a" is the preferred BIDS representation for missing data.